File: ql_lich_trinh_giang_day/app/Models/PhanCong.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class PhanCong extends Model
{
    protected $table = 'PhanCong';
    protected $primaryKey = 'MaPhanCong'; // hoặc 'MaPC' — tùy bảng bạn đặt
    public $timestamps = false;
    protected $fillable = ['MaLHP', 'MaGV', 'NgayPhanCong'];

    public function giangVien()
    {
        return $this->belongsTo(GiangVien::class, 'MaGV', 'MaGV');
    }

    public function lopHocPhan()
    {
        return $this->belongsTo(LopHocPhan::class, 'MaLHP', 'MaLHP');
    }
}

File: ql_lich_giang_day_app/lib/models/phancong.dart
class PhanCong {
  final int maPhanCong;
  final int maLHP;
  final int maGV;
  final String ngayPhanCong;

  PhanCong({
    required this.maPhanCong,
    required this.maLHP,
    required this.maGV,
    required this.ngayPhanCong,
  });

  factory PhanCong.fromJson(Map<String, dynamic> json) => PhanCong(
    maPhanCong: json['MaPhanCong'],
    maLHP: json['MaLHP'],
    maGV: json['MaGV'],
    ngayPhanCong: json['NgayPhanCong'],
  );

  Map<String, dynamic> toJson() => {
    'MaPhanCong': maPhanCong,
    'MaLHP': maLHP,
    'MaGV': maGV,
    'NgayPhanCong': ngayPhanCong,
  };
}
